main.ts

```diff
@@ -217,13 +217,17 @@ class Translator {
         break;
 
       case ts.SyntaxKind.VariableDeclarationList:
+        // Note that a VariableDeclarationList can only occur as part of a for loop.
         var varDeclList = <ts.VariableDeclarationList>node;
-        this.visitList(varDeclList.declarations, ';');
+        this.visitList(varDeclList.declarations, ',');
         break;
 
       case ts.SyntaxKind.VariableDeclaration:
         var varDecl = <ts.VariableDeclaration>node;
-        if (varDecl.type) {
+        if (varDecl.parent.declarations[0] !== varDecl) {
```

Comment:
maybe it'd be more readable to put this check into the `if (varDecl.type)` branch below?